feat: add component register utility (#268)
ithaka/pharos: packages/pharos/README.md

## Registering components

> 1. To allow multiple versions of Pharos to exist on a page, this package only exports component classes for you to register on the [custom element registry](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry) in your application. To register components, you can use the register utility to define all the Pharos components your app/bundle references using `{app/bundle}` as the prefix:

Reviewer: Is this any clearer?

```suggestion
1. To allow multiple versions of Pharos to exist on a page, this package only exports component classes for you to register on the [custom element registry](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry) in your application. To register components, you can use the `registerComponents` utility to define all the Pharos components your application uses, with a consistent scoping prefix:
```

```javascript
import { PharosAlert, PharosButton, PharosIcon } from '@ithaka/pharos';
import registerComponents from '@ithaka/pharos/lib/utils/registerComponents';

registerComponents('{prefix}', [PharosAlert, PharosButton, PharosIcon]);
```

> **Note: Micro frontend remotes and hosts that share Pharos must use the utility to avoid registering elements with the same class instance.**

Reviewer: ```suggestion
**Note: When using module federation, you may only share Pharos successfully by using the `registerComponents` utility. This avoids registering elements with the same underlying class instance, which would result in an error.**
```

> To manually register a component, import the classes you wish to use in your application's entrypoint and define the component with a tag name in the form of `{app/bundle}-pharos-{component}` and a trivial subclass that extends the Pharos class wrapped in the `PharosComponentMixin`:

Reviewer: This is a pretty beefy sentence—wasn't really sure how best to break it up:

```suggestion
To manually register a component, import the classes you wish to use in your application's entrypoint and define the custom element with a tag name in the form of `{app/bundle}-pharos-{component}` and a trivial subclass that extends the Pharos class wrapped in the `PharosComponentMixin`:
```

## Using Pharos components in React

[React](https://reactjs.org/) is a JavaScript library used to build encapsulated components that manage their own state. React doesn't currently play perfectly with native web components, so if you're developing a React application you need to use the wrapper components provided in the `lib/react-components/` directory of the package.

> To use the React components, first register the components using the register utility and then in your app's entry file use the context provider `PharosContext` to indicate the tag name prefix you registered them under:

Reviewer: ```suggestion
To use the React components, first register the components using the `registerComponents` utility. Then, in your app's entry file, use the `PharosContext` context provider to indicate the scoping prefix you registered them under:
```

```javascript
import { PharosContext } from '@ithaka/pharos/lib/utils/PharosContext';

const context = { prefix: 'homepage' };
```

Then, render it in your JSX:

```jsx
<PharosContext.Provider value={context}>...app code</PharosContext.Provider>
```

> To use a component first import it:

Reviewer: ```suggestion
To use a component, first import it:
```
